### ISSUE TITLE
[Bug]: linode_instance_config update doesn't change config, re-apply shows changes everytime

### ISSUE THREAD

**Issue by codedumper** [linode/terraform-provider-linode#917]
### Terraform Version

Terraform v1.5.0 on darwin_arm64

### Linode Provider Version

version = "2.5.0"

### Effected Terraform Resources

linode_instance_config

### Terraform Config Files

terraform {
  required_providers {
    linode = {
      source  = "linode/linode"
      version = "2.5.0"
    }
  }
}

provider "linode" {
}

### Debug Output

TF_LINODE=DEBUG and LINODE_DEBUG=1 doesn't change the output:

```
terraform -chdir=././architecture//test/ apply -var-file=../../terraform.tfvars
module.test.data.linode_profile.me: Reading...
module.test_stackscript.linode_stackscript.stackscript: Refreshing state... [id=1199793]
module.test.linode_instance.instance: Refreshing state... [id=47291484]
module.test.data.linode_profile.me: Read complete after 0s [id={"uid":4327143,"username":"the-croo-group","email":"infrastructure-admin@croo.io","timezone":"GMT","email_notifications":true,"ip_whitelist_enabled":false,"two_factor_auth":true,"restricted":false,"lish_auth_method":"password_keys","referrals":{"total":0,"completed":0,"pending":0,"credit":0,"code":"","url":""},"authorized_keys":null}]
module.test.linode_instance_disk.disks["sdb"]: Refreshing state... [id=94244840]
module.test.linode_instance_disk.disks["sda"]: Refreshing state... [id=94244894]
module.test.linode_instance_config.config: Refreshing state... [id=50165576]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # module.test.linode_instance_config.config will be updated in-place
  ~ resource "linode_instance_config" "config" {
        id           = "50165576"
        # (8 unchanged attributes hidden)

      ~ interface {
          ~ label        = "ca-p01-telephony-core-vlan" -> "telephony-core"
            # (2 unchanged attributes hidden)
        }
      ~ interface {
          ~ label        = "ca-p01-databases-vlan" -> "databases"
            # (2 unchanged attributes hidden)
        }

        # (2 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.test.linode_instance_config.config: Modifying... [id=50165576]
module.test.linode_instance_config.config: Modifications complete after 1s [id=50165576]

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.

Outputs:

test_instance_id = "47291484"
```

### Panic Output

Here is the same operation to TF_LOG_PROVIDER set to DEBUG:

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

2023-06-28T16:12:46.289-0400 [INFO]  provider: configuring client automatic mTLS
2023-06-28T16:12:46.294-0400 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/linode/linode/2.5.0/darwin_arm64/terraform-provider-linode_v2.5.0 args=[.terraform/providers/registry.terraform.io/linode/linode/2.5.0/darwin_arm64/terraform-provider-linode_v2.5.0]
2023-06-28T16:12:46.297-0400 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/linode/linode/2.5.0/darwin_arm64/terraform-provider-linode_v2.5.0 pid=16562
2023-06-28T16:12:46.298-0400 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/linode/linode/2.5.0/darwin_arm64/terraform-provider-linode_v2.5.0
2023-06-28T16:12:46.308-0400 [DEBUG] provider.terraform-provider-linode_v2.5.0: 2023/06/28 16:12:46 [INFO] LINODE_DEBUG being set to true
2023-06-28T16:12:46.317-0400 [INFO]  provider.terraform-provider-linode_v2.5.0: configuring server automatic mTLS: timestamp=2023-06-28T16:12:46.317-0400
2023-06-28T16:12:46.329-0400 [DEBUG] provider: using plugin: version=5
2023-06-28T16:12:46.329-0400 [DEBUG] provider.terraform-provider-linode_v2.5.0: plugin address: address=/var/folders/vr/4znjkf353rq2539zfnjsv2yc0000gn/T/plugin4200212483 network=unix timestamp=2023-06-28T16:12:46.329-0400
2023-06-28T16:12:46.340-0400 [DEBUG] provider.terraform-provider-linode_v2.5.0: Calling provider defined Provider ValidateConfig: tf_rpc=PrepareProviderConfig @caller=github.com/hashicorp/terraform-plugin-framework@v1.3.1/internal/fwserver/server_validateproviderconfig.go:73 tf_provider_addr=registry.terraform.io/linode/linode tf_mux_provider=*proto5server.Server tf_req_id=4f46ef91-ca29-310d-10d8-1a4ada48028b @module=sdk.framework timestamp=2023-06-28T16:12:46.340-0400
2023-06-28T16:12:46.340-0400 [DEBUG] provider.terraform-provider-linode_v2.5.0: Called provider defined Provider ValidateConfig: @module=sdk.framework tf_mux_provider=*proto5server.Server tf_req_id=4f46ef91-ca29-310d-10d8-1a4ada48028b @caller=github.com/hashicorp/terraform-plugin-framework@v1.3.1/internal/fwserver/server_validateproviderconfig.go:75 tf_provider_addr=registry.terraform.io/linode/linode tf_rpc=PrepareProviderConfig timestamp=2023-06-28T16:12:46.340-0400
2023-06-28T16:12:46.340-0400 [DEBUG] provider.terraform-provider-linode_v2.5.0: Calling provider defined Type Validate: @module=sdk.framework tf_attribute_path=lke_event_poll_ms tf_mux_provider=*proto5server.Server tf_req_id=4f46ef91-ca29-310d-10d8-1a4ada48028b tf_rpc=PrepareProviderConfig @caller=github.com/hashicorp/terraform-plugin-framework@v1.3.1/internal/fwschemadata/data_value.go:81 tf_provider_addr=registry.terraform.io/linode/linode timestamp=2023-06-28T16:12:46.340-0400
2023-06-28T16:12:46.340-0400 [DEBUG] provider.terraform-provider-linode_v2.5.0: Called provider defined Type Validate: tf_req_id=4f46ef91-ca29-310d-10d8-1a4ada48028b @caller=github.com/hashicorp/terraform-plugin-framework@v1.3.1/internal/fwschemadata/data_value.go:83 @module=sdk.framework tf_attribute_path=lke_event_poll_ms tf_mux_provider=*proto5server.Server tf_provider_addr=registry.terraform.io/linode/linode tf_rpc=PrepareProviderConfig timestamp=2023-06-28T16:12:46.340-0400
2023-06-28T16:12:46.340-0400 [DEBUG] provider.terraform-provider-linode_v2.5.0: Calling provider defined Type Validate: tf_mux_provider=*proto5server.Server tf_rpc=PrepareProviderConfig @caller=github.com/hashicorp/terraform-plugin-framework@v1.3.1/internal/fwschemadata/data_value.go:81 @module=sdk.framework tf_attribute_path=max_retry_delay_ms tf_provider_addr=registry.terraform.io/linode/linode tf_req_id=4f46ef91-ca29-310d-10d8-1a4ada48028b timestamp=2023-06-28T16:12:46.340-0400
2023-06-28T16:12:46.340-0400 [DEBUG] provider.terraform-provider-linode_v2.5.0: Called provider defined Type Validate: @module=sdk.framework tf_attribute_path=max_retry_delay_ms tf_provider_addr=registry.terraform.io/linode/linode @caller=github.com/hashicorp/terraform-plugin-framework@v1.3.1/internal/fwschemadata/data_value.go:83 tf_mux_provider=*proto5server.Server tf_req_id=4f46ef91-ca29-310d-10d8-1a4ada48028b tf_rpc=PrepareProviderConfig timestamp=2023-06-28T16:12:46.340-0400
2023-06-28T16:12:46.340-0400 [DEBUG] provider.terraform-provider-linode_v2.5.0: Calling provider defined Type Validate: @caller=github.com/hashicorp/terraform-plugin-framework@v1.3.1/internal/fwschemadata/data_value.go:81 @module=sdk.framework tf_attribute_path=min_retry_delay_ms tf_provider_addr=registry.terraform.io/linode/linode tf_req_id=4f46ef91-ca29-310d-10d8-1a4ada48028b tf_mux_provider=*proto5server.Server tf_rpc=PrepareProviderConfig timestamp=2023-06-28T16:12:46.340-0400
2023-06-28T16:12:46.341-0400 [DEBUG] provider.terraform-provider-linode_v2.5.0: Called provider defined Type Validate: @module=sdk.framework tf_provider_addr=registry.terraform.io/linode/linode tf_rpc=PrepareProviderConfig @caller=github.com/hashicorp/terraform-plugin-framework@v1.3.1/internal/fwschemadata/data_value.go:83 tf_attribute_path=min_retry_delay_ms tf_mux_provider=*proto5server.Server tf_req_id=4f46ef91-ca29-310d-10d8-1a4ada48028b timestamp=2023-06-28T16:12:46.340-0400
2023-06-28T16:12:46.341-0400 [DEBUG] provider.terraform-provider-linode_v2.5.0: Calling provider defined Type Validate: tf_provider_addr=registry.terraform.io/linode/linode tf_req_id=4f46ef91-ca29-310d-10d8-1a4ada48028b tf_rpc=PrepareProviderConfig @caller=github.com/hashicorp/terraform-plugin-framework@v1.3.1/internal/fwschemadata/data_value.go:81 @module=sdk.framework tf_attribute_path=event_poll_ms tf_mux_provider=*proto5server.Server timestamp=2023-06-28T16:12:46.340-0400
2023-06-28T16:12:46.341-0400 [DEBUG] provider.terraform-provider-linode_v2.5.0: Called provider defined Type Validate: tf_provider_addr=registry.terraform.io/linode/linode @caller=github.com/hashicorp/terraform-plugin-framework@v1.3.1/internal/fwschemadata/data_value.go:83 @module=sdk.framework tf_attribute_path=event_poll_ms tf_mux_provider=*proto5server.Server tf_req_id=4f46ef91-ca29-310d-10d8-1a4ada48028b tf_rpc=PrepareProviderConfig timestamp=2023-06-28T16:12:46.340-0400
2023-06-28T16:12:46.341-0400 [DEBUG] provider.terraform-provider-linode_v2.5.0: Calling provider defined Type Validate: tf_provider_addr=registry.terraform.io/linode/linode tf_req_id=4f46ef91-ca29-310d-10d8-1a4ada48028b tf_rpc=PrepareProviderConfig @module=sdk.framework @caller=github.com/hashicorp/terraform-plugin-framework@v1.3.1/internal/fwschemadata/data_value.go:81 tf_attribute_path=lke_node_ready_poll_ms tf_mux_provider=*proto5server.Server timestamp=2023-06-28T16:12:46.340-0400
2023-06-28T16:12:46.341-0400 [DEBUG] provider.terraform-provider-linode_v2.5.0: Called provider defined Type Validate: @caller=github.com/hashicorp/terraform-plugin-framework@v1.3.1/internal/fwschemadata/data_value.go:83 tf_attribute_path=lke_node_ready_poll_ms tf_provider_addr=registry.terraform.io/linode/linode tf_req_id=4f46ef91-ca29-310d-10d8-1a4ada48028b tf_rpc=PrepareProviderConfig @module=sdk.framework tf_mux_provider=*proto5server.Server timestamp=2023-06-28T16:12:46.340-0400
2023-06-28T16:12:46.342-0400 [DEBUG] provider.terraform-provider-linode_v2.5.0: Calling provider defined Provider Configure: @caller=github.com/hashicorp/terraform-plugin-framework@v1.3.1/internal/fwserver/server_configureprovider.go:15 @module=sdk.framework tf_provider_addr=registry.terraform.io/linode/linode tf_req_id=a90ab177-c3f6-8e0f-7938-7a74c91f6c06 tf_mux_provider=*proto5server.Server tf_rpc=ConfigureProvider timestamp=2023-06-28T16:12:46.342-0400
2023-06-28T16:12:46.342-0400 [DEBUG] provider.terraform-provider-linode_v2.5.0: 2023/06/28 16:12:46 [INFO] Linode config does not exist, skipping..
2023-06-28T16:12:46.342-0400 [DEBUG] provider.terraform-provider-linode_v2.5.0: Called provider defined Provider Configure: tf_rpc=ConfigureProvider @caller=github.com/hashicorp/terraform-plugin-framework@v1.3.1/internal/fwserver/server_configureprovider.go:23 tf_mux_provider=*proto5server.Server @module=sdk.framework tf_provider_addr=registry.terraform.io/linode/linode tf_req_id=a90ab177-c3f6-8e0f-7938-7a74c91f6c06 timestamp=2023-06-28T16:12:46.342-0400
2023-06-28T16:12:46.342-0400 [DEBUG] provider.terraform-provider-linode_v2.5.0: 2023/06/28 16:12:46 [INFO] Linode config does not exist, skipping..
module.test.linode_instance_config.config: Modifying... [id=50165576]
module.test.linode_instance_config.config: Modifications complete after 1s [id=50165576]
2023-06-28T16:12:47.791-0400 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-06-28T16:12:47.794-0400 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/linode/linode/2.5.0/darwin_arm64/terraform-provider-linode_v2.5.0 pid=16562
2023-06-28T16:12:47.794-0400 [DEBUG] provider: plugin exited

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.

Outputs:

test_instance_id = "47291484"

### Expected Behavior

Config change should update instance as proposed in the plan. 
Depending on change, some change could require reboot of that instance.

Issue is appearing while renaming the vlan. 
Terraform plan shows the update, 

```
  # module.test.linode_instance_config.config will be updated in-place
  ~ resource "linode_instance_config" "config" {
        id           = "50165576"
        # (8 unchanged attributes hidden)

      ~ interface {
          ~ label        = "ca-p01-telephony-core-vlan" -> "telephony-core"
            # (2 unchanged attributes hidden)
        }
      ~ interface {
          ~ label        = "ca-p01-databases-vlan" -> "databases"
            # (2 unchanged attributes hidden)
        }

        # (2 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.test.linode_instance_config.config: Modifying... [id=50165576]
module.test.linode_instance_config.config: Modifications complete after 1s [id=50165576]

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.

Outputs:

test_instance_id = "47291484"

```
But change doesn't appear on Linux Console or instance.
If you re-apply it, the same change is proposed again.

### Actual Behavior

Terraform plan/apply shows the VLAN name in-place update.

Apply shows the modification running, then completed.
None of the change are reflected in console or instance.
tfstate shows the old values.

There is possibly an error returned by provider API that is not reflected into terraform.
This issue is very critical for us. We are not able to change any instances, and it put our deployment on hold. 

### Steps to Reproduce

- Create a linode_instance using instance_config with a VLAN.
- terraform apply
- change the VLAN name in config
- terraform apply
- nothing change in console
- terraform apply
- same change is proposed again

**Comment by codedumper** [comment]
As additional information, it is possible to do the change through UI/Console.
I did the test to confirm this is possible to update instance VLAN configuration.

In our case, all infrastructure is deployed through IaC. We need that to be resolved into the linode provider.

**Comment by lgarber-akamai** [comment]
@codedumper This issue has been resolved in v2.5.1 :tada: 

Let me know if this issue continues to occur after upgrading.

**Comment by codedumper** [comment]
> @codedumper This issue has been resolved in v2.5.1 🎉
> 
> Let me know if this issue continues to occur after upgrading.

I just updated my provider. I confirm configuration is well updated and the re-apply doesn't show anymore the changes.
This quick fix is really appreciated. Thank you @lgarber-akamai  and all the team.
